lib/views/pages/transaction_summary.dart

```diff
@@ -3,24 +3,77 @@ import 'package:intl/intl.dart';
 import 'package:sun_flutter_capstone/consts/global_style.dart';
 import 'package:sun_flutter_capstone/views/widgets/cards/elevated_card.dart';
 import 'package:sun_flutter_capstone/consts/consts.dart';
+import 'package:sun_flutter_capstone/controllers/transactions_controller.dart';
+import 'package:hooks_riverpod/hooks_riverpod.dart';
 
-class TransactionSummary extends StatelessWidget {
-  final double totalBalance;
-  final double totalIncome;
-  final double totalExpenses;
+class TransactionSummary extends StatefulHookConsumerWidget {
   final String currency;
 
   const TransactionSummary({
     Key? key,
-    required this.totalBalance,
-    required this.totalIncome,
-    required this.totalExpenses,
     required this.currency,
   }) : super(key: key);
 
+  @override
+  ConsumerState<ConsumerStatefulWidget> createState() =>
+      _TransactionSummary();
+}
+
+class _TransactionSummary extends ConsumerState<TransactionSummary> {
+
   @override
   Widget build(BuildContext context) {
     final amountFormat = AmountFormat();
+    final transactionState = ref.watch(transactionsNotifierProvider);
+    final expensesState = ref.watch(expenseTransactionsProvider);
+    final incomesState = ref.watch(incomeTransactionsProvider);
+
+    getData(data, section){
+      double _totalExpense = 0;
+      double _totalIncome = 0;
+      for (var dt in data) {
+        if (dt['type'] == 'expense') {
+          _totalExpense += dt['amount'];
+        } else {
+          _totalIncome += dt['amount'];
+        }
+      }
+      if (section == 'expense') return amountFormat.amount(_totalExpense, widget.currency);
+      else if (section == 'income') return amountFormat.amount(_totalIncome, widget.currency);
+      else return amountFormat.amount(_totalIncome - _totalExpense, widget.currency);
```

Comment:
```suggestion
      if (section == 'expense') return amountFormat.amount(_totalExpense, widget.currency);
      if (section == 'income') return amountFormat.amount(_totalIncome, widget.currency);
      
      return amountFormat.amount(_totalIncome - _totalExpense, widget.currency);
```

getting warnings that you should use curly braces. you can also remove "else" keyword
![image](https://user-images.githubusercontent.com/86587091/166202786-d8dba1de-6019-42d5-9a01-a7c9cad7470d.png)